### PR TITLE
Patches for v100

### PR DESCRIPTION
**Pull request by csagan5**
I have not finished building but most of the build issues are solved; the user agent patch needs an update because `content/common/renderer.mojom` changed the way user agent is provided.

@uazo can you please fix that?

Changes:
* [x] fix User-Agent change patch
* [x] sort patches changing PrivacySettings to be closer in the sequence

### CODE REVIEW

**Comment by csagan5** [comment]
Without the user agent patch there is 1 crash that I cannot debug because `stacktrace.py` fails to find symbols; @uazo does it work for you after the export of symbols?

**Comment by uazo** [comment]
> the user agent patch needs an update because content/common/renderer.mojom changed the way user agent is provided.

I changed so:
```
    for (auto iter = content::RenderProcessHost::AllHostsIterator(); !iter.IsAtEnd();
         iter.Advance()) {
      if (iter.GetCurrentValue()->IsInitializedAndNotDead()) {
        std::vector<std::string> cors_exempt_header_list;
        iter.GetCurrentValue()->GetRendererInterface()->InitializeRenderer(       <-----
          /*user_agent*/ ua, /*full_user_agent*/ ua, /*reduced_user_agent*/ ua,
          /*metadata*/ blink::UserAgentMetadata(), cors_exempt_header_list);
      }
    }
```

> does it work for you after the export of symbols?

mine version works, I haven't compared it to yours yet.
I have to carry over the various post build changes to the patches and modify some that are no longer valid, then I check the differences. I think I can make it in the afternoon.

**Comment by csagan5** [comment]
> > does it work for you after the export of symbols?
> 
> mine version works, I haven't compared it to yours yet. I have to carry over the various post build changes to the patches and modify some that are no longer valid, then I check the differences. I think I can make it in the afternoon.

I have actually 2 problems:
* the crash itself
* using `stack.py` is not working

What I get is this:
```
Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  0000000003956a64  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000004ea0bb4  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000004e72b84  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000003756954  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  00000000046aee30  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000003756754  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  00000000037571d8  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000003920678  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000003936480  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  000000000396c10c  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  000000000396c0cc  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  000000000396bf3c  offset 0x15bd000                                                                  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
  0000000000014ff8  android::Looper::pollInner(int)+836                                               /system/lib64/libutils.so
  0000000000014c18  android::Looper::pollOnce(int, int*, int*, void**)+60                             /system/lib64/libutils.so
  00000000001262d0  android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44  /system/lib64/libandroid_runtime.so
  00000000003e5fec  offset 0x3d1000) (android.media.MediaExtractor.seekTo [DEDUPED]+140               /system/framework/arm64/boot-framework.oat
  0000000000ab8204  offset 0x3d1000) (android.os.MessageQueue.next+244                                /system/framework/arm64/boot-framework.oat
  0000000000ab5698  offset 0x3d1000) (android.os.Looper.loop+776                                      /system/framework/arm64/boot-framework.oat
  0000000000884378  offset 0x3d1000) (android.app.ActivityThread.main+664                             /system/framework/arm64/boot-framework.oat
  000000000055604c  art_quick_invoke_static_stub+604                                                  /system/lib64/libart.so
  00000000000cf8e8  art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232  /system/lib64/libart.so
  000000000045d04c  art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104  /system/lib64/libart.so
  000000000045eaa0  art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1440  /system/lib64/libart.so
  00000000003ee9e0  art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+52             /system/lib64/libart.so
  000000000011e6d4  offset 0x114000) (java.lang.Class.getDeclaredMethodInternal [DEDUPED]+180         /system/framework/arm64/boot-core-oj.oat
  0000000000bed798  offset 0x3d1000) (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+136  /system/framework/arm64/boot-framework.oat
  0000000000bf4980  offset 0x3d1000) (com.android.internal.os.ZygoteInit.main+3088                    /system/framework/arm64/boot-framework.oat
  000000000055604c  art_quick_invoke_static_stub+604                                                  /system/lib64/libart.so
  00000000000cf8e8  art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232  /system/lib64/libart.so
  000000000045d04c  art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104  /system/lib64/libart.so
  000000000045ccac  art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+424  /system/lib64/libart.so
  000000000036236c  art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+652  /system/lib64/libart.so
  00000000000b1844  _JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+116                     /system/lib64/libandroid_runtime.so
  00000000000b4268  android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+752  /system/lib64/libandroid_runtime.so
  000000000000251c  main+2000                                                                         /system/bin/app_process64
  00000000000ca4f4  __libc_init+88                                                                    /system/lib64/libc.so
```

And before that:
```
pid: 16657, tid: 16657, name: bromite.bromite  >>> org.bromite.bromite <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x18
DEBUG:root:Found trace line: 03-27 22:17:13.138 16832 16832 F DEBUG   :     #00 pc 0000000003956a64  /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so (offset 0x15bd000)
DEBUG:root:Identified lib: /data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
DEBUG:root:TranslateLibPath: lib=/data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so library_name=libchrome.so
DEBUG:root:GetCandidates: prefiltered candidates = [u'out/Release_arm64/lib.unstripped/libchrome.so']
DEBUG:root:TranslateLibPath: candidate_libraries=[]
ERROR:root:Can't run llvm-symbolizer! Given binary is not a valid target. path=/data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so
DEBUG:root:symbol information: [('<UNKNOWN>', u'/data/app/org.bromite.bromite-2mfXhANMBGaaVm4UpgimjA==/libchrome.so')]
```

I think it is showing `offset 0x15bd000` because it cannot find the symbols, but it should?
Unless I am doing something wrong with the invocation of `stack.py`: how do you invoke it with the separate symbols? (like those released in the `_debug.7z` archive)

**Comment by uazo** [comment]
> using stack.py is not working

Try this:
```
TOMBSTONE=tombstone_01 && adb pull /data/tombstones/$TOMBSTONE && \
    third_party/android_platform/development/scripts/stack \ 
         --output-directory out/bromite/ ../../$TOMBSTONE >../../$TOMBSTONE.txt
```

**Comment by uazo** [comment]
there are not many different between mine and your version, I can't explain the crash except for the fact that in debug I don't have the same gn args, I have to try to compile not in debug.

but I noticed this:
```
+#if BUILDFLAG(SAFE_BROWSING_AVAILABLE)
   safe_browsing::file_type::RegisterProfilePrefs(registry);
   safe_browsing::RegisterProfilePrefs(registry);
+#endif
```
`safe_browsing::RegisterProfilePrefs(registry);` it's better to be out of the `#if` because the preferences have to be registered otherwise it crashes at the first request, maybe that's it.

**Comment by csagan5** [comment]
> Try this:
> 
>> TOMBSTONE=tombstone_01 && adb pull /data/tombstones/$TOMBSTONE

I used `adb bugreport bugreport.zip` and I could access the latest tombstone, however I still get no symbols for that `offset 0x15bd000`. I see other thread stack traces now but they are all waiting on conditions or still obfuscated.

> `safe_browsing::RegisterProfilePrefs(registry);` it's better to be out of the `#if` because the preferences have to be registered otherwise it crashes at the first request, maybe that's it.

Good idea, thanks; I also thought it could crash when making those changes. I will try to change that part.

**Comment by uazo** [comment]
how you solved [Add option to force tablet UI](https://github.com/uazo/bromite/blob/next/v100/build/patches/Add-option-to-force-tablet-UI.patch)?

**Comment by csagan5** [comment]
> how you solved [Add option to force tablet UI](https://github.com/uazo/bromite/blob/next/v100/build/patches/Add-option-to-force-tablet-UI.patch) settings?

What issue, specifically?

File: build/patches/Disable-Accessibility-service-by-default.patch [hunk 1]
```diff
@@ -3,46 +3,60 @@ Date: Mon, 8 Nov 2021 09:47:23 +0000
...
+         android:title="@string/font_size"
+diff --git a/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java b/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
+--- a/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
++++ b/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
+@@ -17,6 +17,9 @@ import org.chromium.components.browser_ui.accessibility.FontSizePrefs.FontSizePr
+ import org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference;
+ import org.chromium.components.browser_ui.settings.SettingsUtils;
+ 
++import org.chromium.chrome.browser.preferences.ChromePreferenceKeys;
++import org.chromium.chrome.browser.preferences.SharedPreferencesManager;
```
[the remark this answers is not in the document]
Comment (csagan5) [reply build/patches/Disable-Accessibility-service-by-default.patch:51]: They cannot, they are needed for the "force tablet UI" patch.

**Comment by uazo** [comment]
> What issue, specifically?

ah ok, I understand, then it's my bottom toolbar's fault that needs `Snackbar`
forget what I asked you.

File: build/patches/Disable-Accessibility-service-by-default.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (uazo) [reply build/patches/Disable-Accessibility-service-by-default.patch:51]: Ok, you are right

File: build/patches/Allow-building-without-enable_reporting.patch [hunk 2]
```diff
@@ -3,35 +3,48 @@ Date: Fri, 22 May 2020 22:43:27 -0400
...
+ 17 files changed, 51 insertions(+), 39 deletions(-)
 
+diff --git a/content/browser/browser_interface_binders.cc b/content/browser/browser_interface_binders.cc
+--- a/content/browser/browser_interface_binders.cc
++++ b/content/browser/browser_interface_binders.cc
+@@ -164,7 +164,6 @@
+ #include "content/browser/renderer_host/render_widget_host_view_android.h"
+ #include "services/device/public/mojom/nfc.mojom.h"
+ #include "third_party/blink/public/mojom/hid/hid.mojom.h"
+-#include "third_party/blink/public/mojom/unhandled_tap_notifier/unhandled_tap_notifier.mojom.h"
```
[the remark this answers is not in the document]
Comment (uazo) [reply build/patches/Allow-building-without-enable_reporting.patch:32]: yes

[the remark this answers is not in the document]
Comment (csagan5) [reply build/patches/Allow-building-without-enable_reporting.patch:32]: Ok, I will adopt it; thanks!

**Comment by csagan5** [comment]
> there are not many different between mine and your version

Note, all patches changing the `PrivacySettings` are now closer in their sequence; hopefully this will help when porting patches. I do not plan to merge those but if you see other patches that could be merged please let me know.

**Review by uazo — APPROVED** [review]

**Comment by uazo** [comment]
@csagan5 you can see the [new flags of v100](https://github.com/uazo/bromite-flags-list/blob/main/excel/flags-list-100.0.4896.46.xlsx)

personally I would activate `kAutomaticLazyFrameLoadingToAds` and `kIncognitoParamFilterEnabled` (toh, they added url cleanup)

`kOpaqueResponseBlockingV01` must be better understood, and in my opinion it should be seen exactly how [chips](https://github.com/WICG/CHIPS) work, because as it is it should not be activated (see `kClientHintsPartitionedCookies` and `kPartitionedCookies`)

I also really like the `kIsolateSandboxedIframes`, but to be modified in order to inhibit javascript in the iframes ...

**Comment by csagan5** [comment]
Yes they seem interesting; I can take note to enable some of these for next release as it is late now for v100.

> `kOpaqueResponseBlockingV01` (see https://bugs.chromium.org/p/chromium/issues/detail?id=1178928) must be better understood

If it is useful as a sort of offline cache and this option is enabled, the offline cache functionality is gone?

> personally I would activate `kAutomaticLazyFrameLoadingToAds` and `kIncognitoParamFilterEnabled` (toh, they added url cleanup)

`kAutomaticLazyFrameLoadingToAds` can be enabled but `kIncognitoParamFilterEnabled` would need a flag?

> and in my opinion it should be seen exactly how [chips](https://github.com/WICG/CHIPS) work, because as it is it should not be activated (see `kClientHintsPartitionedCookies` and `kPartitionedCookies`)

I know nothing about CHIPS.

> I also really like the `kIsolateSandboxedIframes`, but to be modified in order to inhibit javascript in the iframes ...

I would also enabled this because it has advantages against side-channel attacks.

**Comment by uazo** [comment]
> If it is useful as a sort of offline cache 

sure? it is not the evolution of https://www.chromium.org/Home/chromium-security/corb-for-developers/

> this option is enabled,

No, it's disabled.

> kIncognitoParamFilterEnabled would need a flag

it would be better

> I know nothing about CHIPS.

study it because in my opinion in a while they will remove the flags and understand afterwards how to deactivate it will become complex.

> I would also enabled this because it has advantages against side-channel attacks.

as it is no, let's talk about it first

**Comment by uazo** [comment]
> but "opaque responses" were born for this use-case? https://stackoverflow.com/a/36303436

could be. but chromium blocks replies with an empty message instead of replying with a network error, contrary to the initial specification (https://bugs.chromium.org/p/chromium/issues/detail?id=1178928#c16)

> offline cache functionality is gone?

no, but if you want it is more restrictive.

**Comment by csagan5** [comment]
Ok, so they are making it more compliant to the latest ORB spec now?

> > offline cache functionality is gone?
> 
> no, but if you want it is more restrictive.

Ok, then we are possibly going to see it enabled by upstream as well, later on. It could be not enabled by default because they are ironing out bugs.

**Comment by csagan5** [comment]
`kClientHintsPartitionedCookies` is currently a placeholder, see https://bugs.chromium.org/p/chromium/issues/detail?id=1225444

I am thinking to enable `kIsolateSandboxedIframes` and `kAutomaticLazyFrameLoadingToAds` by default.

**Comment by uazo** [comment]
for `kIsolateSandboxedIframes` I had signed up to check [ParseWebSandboxPolicy()](https://github.com/uazo/bromite-flags-list/blob/main/flags-to-check#L152), although you don't remember why. I will let you know.

**Comment by uazo** [comment]
for `kClientHintsPartitionedCookies` my fear is that it is only active if third-party cookies are active, and in my opinion it is better to reject them rather than partition them.

**Comment by csagan5** [comment]
> for `kClientHintsPartitionedCookies` my fear is that it is only active if third-party cookies are active, and in my opinion it is better to reject them rather than partition them.

If the browser currently supports accepting them then that feature should be improved, privacy-wise.

**Pull request merged**